ingest.py

```
@@ -0,0 +1,310 @@
+import sys
```

More a general comment: The error handling seems a little fragile. In case something fails the script usually just continues and ignores the errors. This might leave your directory in an inconsistent state from which it might be hard to recover. Maybe you want to instead record all operations that you want to do and only apply them if everything worked correctly until then.